firefox_ui_tests/remote/security/test_untrusted_connection_error_page.py

```diff
@@ -0,0 +1,35 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import time
+
+from marionette_driver import By
+from marionette_driver.errors import MarionetteException
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestUntrustedConnectionErrorPage(FirefoxTestCase):
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        self.test_url = 'https://ssl-selfsigned.mozqa.com'
+
+    def tearDown(self):
+        FirefoxTestCase.tearDown(self)
+
+    def test_untrusted_connection_error_page(self):
+        # Test the GetMeOutOfHereButton from an Untrusted Error page
+        with self.marionette.using_context('content'):
+            self.assertRaises(MarionetteException,
+                              self.marionette.navigate, self.test_url)
+            # Wait for the DOM to receive events
+            time.sleep(1)
+
+            button = self.marionette.find_element(
+                By.ID, "getMeOutOfHereButton")
+            self.assertTrue(button.is_displayed())
```

Comment:
updated the PR with the feedback!

On Thu, Mar 19, 2015 at 8:15 AM, Henrik Skupin <notifications@github.com>
wrote:

> In
> firefox_ui_tests/remote/security/test_untrusted_connection_error_page.py
> <https://github.com/mozilla/firefox-ui-tests/pull/133#discussion_r26748213>
> :
>
> > +        self.test_url = 'https://ssl-selfsigned.mozqa.com'
> > +
> > +    def tearDown(self):
> > +        FirefoxTestCase.tearDown(self)
> > +
> > +    def test_untrusted_connection_error_page(self):
> > +        # Test the GetMeOutOfHereButton from an Untrusted Error page
> > +        with self.marionette.using_context('content'):
> > +            self.assertRaises(MarionetteException,
> > +                              self.marionette.navigate, self.test_url)
> > +            # Wait for the DOM to receive events
> > +            time.sleep(1)
> > +
> > +            button = self.marionette.find_element(
> > +                By.ID, "getMeOutOfHereButton")
> > +            self.assertTrue(button.is_displayed())
>
> I don't think this is necessary. find_element will fail with an exception
> if the button does not exist. That is what our Mozmill test is covering.
> The case when it is hidden we can get if the click event is not working.
> Also as we have seen lately is_displayed is not reliable and should not
> be used. So lets simply remove this line until we have a proper method to
> check for the displayed status.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/firefox-ui-tests/pull/133/files#r26748213>.
>